api/v2/db/ldap/directory-dao.js

```diff
@@ -0,0 +1,126 @@
+const _ = require('lodash');
+const { serializeDirectories, serializeDirectory } = require('../../serializers/directory-serializer');
+
+const conn = require('./connection');
+
+/**
+ * @summary Map endpoint query to ldap query
+ * @function
+ * @returns {string} string representing search filter for ldap query
+ */
+const mapQuery = (endpointQuery) => {
+  const keyMap = {
+    fuzzyName: 'cn',
+    lastName: 'sn',
+    firstName: 'givenName',
+    primaryAffiliation: 'osuPrimaryAffiliation',
+    onid: 'uid',
+    emailAddress: 'mail',
+    officePhoneNumber: 'telephoneNumber',
+    alternatePhoneNumber: 'osuAltPhoneNumber',
+    faxNumber: 'facsimileTelephoneNumber',
+    phoneNumber: 'telephoneNumber',
+    officeAddress: 'osuOfficeAddress',
+    department: 'osuDepartment',
+  };
+
+  const valueOperations = (key, value) => {
+    const defaultOperation = `${keyMap[key]}=*${value}*`;
+    switch (key) {
+      case 'fuzzyName': {
+        let fuzzyFilters = '|'; // 'or' condition for all name orderings
+
+        const valueTerms = value.split(/[ ,]+/);
+        _.forEach(_.range(valueTerms.length), (commaIndex) => {
+          let firstName = `${valueTerms.slice(0, commaIndex).join(' ')}`;
+          let lastName = `${valueTerms.slice(commaIndex).join(' ')}`;
+
+          // Add leading wildcard for non-null first and last names
+          firstName = (firstName ? `*${firstName}` : firstName);
+          lastName = (lastName ? `*${lastName}` : lastName);
+
+          // Consider first, last ordering and last, first ordering with trailing wildcard
+          fuzzyFilters += `(${keyMap[key]}=${firstName}*, ${lastName}*)`;
+          fuzzyFilters += `(${keyMap[key]}=${lastName}*, ${firstName}*)`;
```

Comment:
`keyMap[key]` is used many times in this block. Can we store it into a var and reuse it?